torch/_inductor/lowering.py

```diff
@@ -400,9 +400,9 @@ def broadcast_symbolic_shapes(a, b):
     for x, y in itertools.zip_longest(
         reversed(a), reversed(b), fillvalue=sympy.Integer(1)
     ):
-        if y == 1:
+        if V.graph.sizevars.shape_env.evaluate_expr(sympy.Eq(y, 1)):
```

Comment:
@ezyang from running my test locally with `TORCH_LOGS="+dynamic"`, this does seem to add extra guards into the shape env. Do you think that's worth worrying about? (or do we have some guard deduping logic later on?)

Reply:
We're supposed to dedupe it! So clean it up into a test case and post a bug about it

Reply:
oh nope nvm - the `guard_added` log prints twice but we don't actually end up with duplicate guards in dynamo